lib/phoenix/router/resource.ex

```diff
@@ -35,13 +35,15 @@ defmodule Phoenix.Router.Resource do
     name     = Keyword.get(options, :name, Phoenix.Naming.resource_name(controller, "Controller"))
     as       = Keyword.get(options, :as, name)
     singular = Keyword.get(options, :singular)
+    private  = Keyword.get(options, :private, %{})
     actions  = extract_actions(options, singular)
 
-    collection  = [path: path, as: as]
+    route       = [as: as, private: private]
+    collection  = [path: path, as: as, private: private]
     member_path = if singular, do: path, else: Path.join(path, ":#{name}_#{param}")
-    member      = [path: member_path, as: as, alias: alias]
+    member      = [path: member_path, as: as, alias: alias, private: private]
```

Comment:
This is a commit I pushed to master and somehow ended up here. We need to be careful when merging!